service/share/cache_availability.go

```diff
@@ -0,0 +1,58 @@
+package share
+
+import (
+	"context"
+
+	"github.com/ipfs/go-datastore"
+	"github.com/ipfs/go-datastore/autobatch"
+	"github.com/ipfs/go-datastore/namespace"
+
+	"github.com/celestiaorg/celestia-node/header/store"
+)
+
+var cacheAvailabilityPrefix = datastore.NewKey("sampling_result")
+
+func rootKey(root *Root) datastore.Key {
+	return datastore.NewKey(root.String())
+}
+
+// cacheAvailability wraps a given Availability (whether it's light or full)
+// and stores the results of a successful sampling routine over a given Root's hash
+// to disk.
+type cacheAvailability struct {
+	avail Availability
+
+	ds *autobatch.Datastore
+}
+
+// NewCacheAvailability wraps the given Availability with an additional datastore
+// for sampling result caching.
+func NewCacheAvailability(avail Availability, ds datastore.Batching) Availability {
+	ds = namespace.Wrap(ds, cacheAvailabilityPrefix)
+	autoDS := autobatch.NewAutoBatching(ds, store.DefaultWriteBatchSize)
```

Comment:
We should not make share pkg depend on the header store pkg. Also, it is totally fine to make those values independent and configured separately.

Reply:
https://github.com/celestiaorg/celestia-node/pull/786/commits/671c56d1a13ed1fb737c3c55ee9f695946b14193

wdyt?

Reply:
We should not share those values. They will be eventually defined independently in the config.